### PR TITLE
[Tensor] Quantized Tensor (Int 8) with Scale

### PR DESCRIPTION
**Pull request by djeong20**
- Quantized Tensor is now present with Int 8 with scale.
- Dequantization is performed by multiplying values by a scaling factor for channels.
- Only read, write, and dequantization operations are allowed.

**Self evaluation:**
1. Build test:   [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/tensor.cpp [hunk 1]
```diff
@@ -2644,6 +2668,51 @@ void Tensor::print(std::ostream &out) const {
 #else
     throw std::invalid_argument("Error: enable-fp16 is not enabled");
 #endif
+  } else if (getDataType() == ml::train::TensorDim::DataType::QINT8) {
+    const int8_t *data = getData<int8_t>();
+    unsigned int len = size();
+    out << "data addr: " << (float *)data << '\n';
```
Comment (DonghakPark) [inline nntrainer/tensor/tensor.cpp:2674]: Avoid using C-style casts ```(float *)data``` Instead, ```reinterpret_cast<float*>(data)``` to make the cast explicit and avoid potential issues with type safety.
```suggestion
    out << "data addr: " << reinterpret_cast<const float*>(data) << '\n';
```

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: api/ccapi/include/tensor_dim.h [hunk 2]
```diff
@@ -48,12 +48,13 @@ class TensorDim {
...
   /**
-   * @brief Tensor Data Type. Currently FP16 & FP32 Support
+   * @brief Tensor Data Type. Currently QINT8, FP16 & FP32 Support
    *
    */
   enum class DataType {
-    FP16, /** half precion */
-    FP32  /** single precision */
+    QINT8, /** quantized int 8*/
+    FP16,  /** half precion */
```
[the remark this answers is not in the document]
Comment (djeong20) [reply api/ccapi/include/tensor_dim.h]: Fixed!

File: nntrainer/tensor/tensor.cpp (hunk 1 above)
Comment (djeong20) [reply nntrainer/tensor/tensor.cpp]: Applied!

**Review by Hyoungsukim — APPROVED** [review]
lgtm

**Review by songgot — APPROVED** [review]
LGTM

**Pull request merged**